docs/design/kubeadm-api.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+type MasterConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Data shared between phases. Used for defaulting
+	Master     Master     `json:"master"`
+	Networking Networking `json:"networking"`
+	CertificatesDir string `json:"certificatesDir"`
+
+	// Phases
+	Certificates     Certificates     `json:"certificates"`
+	KubeConfig       KubeConfig       `json:"kubeConfig"`
+	ControlPlane     ControlPlane     `json:"controlPlane"`
+	Discovery        Discovery        `json:"discovery"`
+	APIConfiguration APIConfiguration `json:"apiConfiguration"`
+	Addons           Addons           `json:"addons"`
+}
+
+type Phase struct {
+	Annotations map[string]string `json:"annotations"`
+}
+
+// Shared
+// TODO: This field should be HA friendly
+type Master struct {
```

Comment:
yes, that stays the same.

This Master struct that's here was named API before, but should not exist at all I think